### ISSUE TITLE
SegmentNotFoundException when using S3 TransferManager to multiPart upload files, tracing by AWS XRay

### ISSUE THREAD

**Issue by Zhenye-Na** [aws/aws-sdk-java-v2#3217]
### Describe the bug

Existing rasied issues:

- https://github.com/aws/aws-xray-sdk-java/issues/313
- https://github.com/aws/aws-sdk-java/issues/1572

Basically the problem is that `transferManager` manages its own thread pool and XRay is not able to gather context for the threads that `transferManager` created. In this case, `SegmentNotFoundException` has been raised

### Expected Behavior

TransferManager should be compatible with XRay for request tracing

### Current Behavior

`SegmentNotFoundException` is raised

### Reproduction Steps

S3 Multipart Upload High Level API - https://docs.aws.amazon.com/AmazonS3/latest/userguide/mpu-upload-object.html

```java

import com.amazonaws.AmazonServiceException;
import com.amazonaws.SdkClientException;
import com.amazonaws.auth.profile.ProfileCredentialsProvider;
import com.amazonaws.regions.Regions;
import com.amazonaws.services.s3.AmazonS3;
import com.amazonaws.services.s3.AmazonS3ClientBuilder;
import com.amazonaws.services.s3.transfer.TransferManager;
import com.amazonaws.services.s3.transfer.TransferManagerBuilder;
import com.amazonaws.services.s3.transfer.Upload;

import java.io.File;

public class HighLevelMultipartUpload {

    public static void main(String[] args) throws Exception {
        Regions clientRegion = Regions.DEFAULT_REGION;
        String bucketName = "*** Bucket name ***";
        String keyName = "*** Object key ***";
        String filePath = "*** Path for file to upload ***";

        try {
            AmazonS3 s3Client = AmazonS3ClientBuilder.standard()
                    .withRegion(clientRegion)
                    .withCredentials(new ProfileCredentialsProvider())
                    .build();
            TransferManager tm = TransferManagerBuilder.standard()
                    .withS3Client(s3Client)
                    .build();

            // TransferManager processes all transfers asynchronously,
            // so this call returns immediately.
            Upload upload = tm.upload(bucketName, keyName, new File(filePath));
            System.out.println("Object upload started");

            // Optionally, wait for the upload to finish before continuing.
            upload.waitForCompletion();
            System.out.println("Object upload complete");
        } catch (AmazonServiceException e) {
            // The call was transmitted successfully, but Amazon S3 couldn't process 
            // it, so it returned an error response.
            e.printStackTrace();
        } catch (SdkClientException e) {
            // Amazon S3 couldn't be contacted for a response, or the client
            // couldn't parse the response from Amazon S3.
            e.printStackTrace();
        }
    }
}

```

and on top of this, using XRay to trace the request

### AWS Java SDK version used

AWSS3JavaClient = 1.12.x

### JDK version used

JDK 8

### Operating System and version

Amazon Linux 2

**Comment by debora-ito** [comment]
Hi @Zhenye-Na thank you for reaching out.

We got the same issue reported but for X-Ray and the SDK client-side metrics here: https://github.com/aws/aws-sdk-java-v2/issues/2919. As it's mentioned in the comments, it's not something we can fix on the SDK side without X-Ray support.

We've opened an issue in the X-Ray SDK repo and we're tracking progress over there: https://github.com/aws/aws-xray-sdk-java/issues/330.

Let us know if this helps.
